marlowe-playground-client/src/SimulationPage/State.purs

```diff
@@ -303,22 +262,16 @@ editorSetValue contents = void $ query _simulatorEditorSlot unit (Monaco.SetText
 editorGetValue :: forall state action msg m. HalogenM state action ChildSlots msg m (Maybe String)
 editorGetValue = query _simulatorEditorSlot unit (Monaco.GetText identity)
 
--- QUESTION: What is the purpose of this function? Why do we need to store the contents of
---           the editor as an _oldContract?
-saveInitialState :: forall m. MonadEffect m => HalogenM State Action ChildSlots Void m Unit
-saveInitialState = do
-  oldContract <- editorGetValue
-  modifying _oldContract
-    ( \x -> case x of
-        Nothing -> Just $ fromMaybe "" oldContract
-        _ -> x
-    )
-
-resetContract :: forall m. HalogenM State Action ChildSlots Void m Unit
-resetContract = do
-  newContract <- editorGetValue
-  modify_
-    ( set _marloweState (NEL.singleton emptyMarloweState)
-        <<< set _oldContract Nothing
+updateContractInEditor ::
+  forall m.
+  MonadAff m =>
+  MonadAsk Env m =>
+  HalogenM State Action ChildSlots Void m Unit
+updateContractInEditor = do
+  executionState <- use (_currentMarloweState <<< _executionState)
+  editorSetValue
+    ( case executionState of
+        SimulationRunning runningState -> show $ genericPretty runningState.contract
+        SimulationNotStarted notStartedState -> maybe "No contract" (show <<< genericPretty) notStartedState.extendedContract
```

Comment:
good idea to have this text although I assume we should never be able to get here, no?

Reply:
Yes, that's right. It is just there so that it will make debugging easier if something goes wrong